From: Kel Modderman <kel@otaku42.de>
Date: Fri, 5 Apr 2019 20:32:40 +0000
Subject: only emit warnings about overriding LSB default with symlink

Bug-Debian: bugs.debian.org/668559

state for the scripts in insserv arg list. Use "overrides" instead of
"overwrites" and define defaults as "LSB defaults".
Bug-Debian: bugs.debian.org/668559
---
 insserv.c | 14 ++++++++------
 1 file changed, 8 insertions(+), 6 deletions(-)

diff --git a/insserv.c b/insserv.c
index 1fffd72..f48bc68 100644
--- a/insserv.c
+++ b/insserv.c
@@ -3393,7 +3393,7 @@ int main (int argc, char *argv[])
 			     */
 			    if (!defaults && (deflvls != service->start->lvl)) {
 				if (!del && isarg && !(argr[curr_argc]))
-				    warn("warning: current start runlevel(s) (%s) of script `%s' overwrites defaults (%s).\n",
+				    warn("warning: current start runlevel(s) (%s) of script `%s' overrides LSB defaults (%s).\n",
 					 service->start->lvl ? lvl2str(service->start->lvl) : "empty", d->d_name, lvl2str(deflvls));
 			    }
 			} else
@@ -3412,8 +3412,9 @@ int main (int argc, char *argv[])
 			     * of the current script.
 			     */
 			    if (!defaults && service->start->lvl != 0) {
-				warn("warning: current start runlevel(s) (%s) of script `%s' overwrites defaults (empty).\n",
-				     lvl2str(service->start->lvl), d->d_name);
+				if (!del && isarg && !(argr[curr_argc]))
+				    warn("warning: current start runlevel(s) (%s) of script `%s' overrides LSB defaults (empty).\n",
+					 lvl2str(service->start->lvl), d->d_name);
 				script_inf.default_start = lvl2str(service->start->lvl);
 			    }
 			}
@@ -3455,7 +3456,7 @@ int main (int argc, char *argv[])
 			     */
 			    if (!defaults && (deflvlk != service->stopp->lvl)) {
 				if (!del && isarg && !(argr[curr_argc]))
-				    warn("warning: current stop runlevel(s) (%s) of script `%s' overwrites defaults (%s).\n",
+				    warn("warning: current stop runlevel(s) (%s) of script `%s' overrides LSB defaults (%s).\n",
 					 service->stopp->lvl ? lvl2str(service->stopp->lvl) : "empty", d->d_name, lvl2str(deflvlk));
 			    }
 			} else
@@ -3474,8 +3475,9 @@ int main (int argc, char *argv[])
 			     * of the current script.
 			     */
 			    if (!defaults && service->stopp->lvl != 0) {
-				warn("warning: current stop runlevel(s) (%s) of script `%s' overwrites defaults (empty).\n",
-				     lvl2str(service->stopp->lvl), d->d_name);
+				if (!del && isarg && !(argr[curr_argc]))
+				    warn("warning: current stop runlevel(s) (%s) of script `%s' overrides LSB defaults (empty).\n",
+					 lvl2str(service->stopp->lvl), d->d_name);
 				script_inf.default_stop = lvl2str(service->stopp->lvl);
 			    }
 			}
